ipaserver/install/httpinstance.py

```diff
@@ -577,3 +581,17 @@ def start_tracking_certificates(self):
         db = certs.CertDB(self.realm, nssdir=paths.HTTPD_ALIAS_DIR)
         db.track_server_cert(self.cert_nickname, self.principal,
                              db.passwd_fname, 'restart_httpd')
+
+    def request_service_keytab(self):
+        super(HTTPInstance, self).request_service_keytab()
+
+        if self.master_fqdn is not None:
```

Comment:
Maybe we can factor this check into a separate method and push it up to the parent class to always call it in `request_service_keytab`. We will thus have a generic solution for all service installers.